container/build_tar.py

```diff
@@ -222,18 +223,34 @@ def write_temp_file(self, data, suffix='tar', mode='wb'):
       os.remove(tmpfile)
 
   def add_pkg_metadata(self, metadata_tar, deb):
+    """
+    Extract the package ``control`` metadata file from a Debian `metadata_tar`
+    tarball file to the status.d directory. Also extract the ``md5sums`` files
+    list file if present.
+    """
     try:
       with tarfile.open(metadata_tar) as tar:
+        tar_members = tar.getmembers()
         # Metadata is expected to be in a file.
-        control_file_member = list(filter(lambda f: os.path.basename(f.name) == TarFile.PKG_METADATA_FILE, tar.getmembers()))
+        control_file_member = list(filter(lambda f: os.path.basename(f.name) == TarFile.PKG_METADATA_FILE, tar_members))
         if not control_file_member:
-           raise self.DebError(deb + ' does not Metadata File!')
+           raise self.DebError(deb + ' does not contain a control Metadata File!')
         control_file = tar.extractfile(control_file_member[0])
         metadata = b''.join(control_file.readlines())
```

Comment:
It feels like this could be more readably expressed as
```
metadata = controlfile.read().decode("utf-8")
pkg_name = TarFile.parse_pkg_name(metadata, deb)
```

Reply:
@aiuto good point. Done in the latest push. I also rebased. Thank you for the review!